packages/backend/src/tenants/service.test.ts

```diff
@@ -0,0 +1,366 @@
+import assert from 'assert'
+import { faker } from '@faker-js/faker'
+import { IocContract } from '@adonisjs/fold'
+import nock from 'nock'
+import { AppServices } from '../app'
+import { initIocContainer } from '..'
+import { createTestApp, TestContainer } from '../tests/app'
+import { TenantService } from './service'
+import { Config, IAppConfig } from '../config/app'
+import { truncateTables } from '../tests/tableManager'
+import { ApolloClient, NormalizedCacheObject } from '@apollo/client'
+import { Tenant } from './model'
+
+describe('Tenant Service', (): void => {
+  let deps: IocContract<AppServices>
+  let appContainer: TestContainer
+  let tenantService: TenantService
+  let config: IAppConfig
+  let apolloClient: ApolloClient<NormalizedCacheObject>
+
+  beforeAll(async (): Promise<void> => {
+    deps = initIocContainer(Config)
+    appContainer = await createTestApp(deps)
+    tenantService = await deps.use('tenantService')
+    config = await deps.use('config')
+    apolloClient = await deps.use('apolloClient')
+  })
+
+  afterEach(async (): Promise<void> => {
+    await truncateTables(appContainer.knex)
+  })
+
+  afterAll(async (): Promise<void> => {
+    nock.cleanAll()
+    await appContainer.shutdown()
+  })
+
+  describe('get', (): void => {
+    test('can get a tenant', async (): Promise<void> => {
+      const createOptions = {
+        apiSecret: 'test-api-secret',
+        publicName: 'test tenant',
+        email: faker.internet.email(),
+        idpConsentUrl: faker.internet.url(),
+        idpSecret: 'test-idp-secret'
+      }
+
+      const createScope = nock(config.authAdminApiUrl)
+        .post('')
+        .reply(200, { data: { createTenant: { id: 1234 } } })
+
+      const createdTenant = await tenantService.create(createOptions)
+      createScope.done()
+
+      const getScope = nock(config.authAdminApiUrl)
+        .post('')
+        .reply(200, {
+          data: {
+            getTenant: {
+              tenant: {
+                idpConsentUrl: createOptions.idpConsentUrl,
+                idpSecret: createOptions.idpSecret
+              }
+            }
+          }
+        })
+      const apolloSpy = jest.spyOn(apolloClient, 'query')
+      const tenant = await tenantService.get(createdTenant.id)
+      assert.ok(tenant)
+      expect(tenant.id).toEqual(createdTenant.id)
```

Comment:
Maybe just 
```
expect(tenant).toEqual(createdTenant)
```
? Don't know if this works :D

Reply:
It breaks because the `updatedAt` for one will always be different from the other, sadly.

Reply:
Why would updatedAt field be different ? You are creating a tenant and getting back data, and then you are querying the exact same entity ... so it should be the same one? Or am I missing something here?

Reply:
Nevermind that, I conflated that result with a different test. This will work, but `createdTenant` just needs to be augmented with the `idpSecret` and `idpConsentUrl`.